pkg/devicemapper/devmapper.go

```diff
@@ -361,15 +361,13 @@ func RemoveDeviceDeferred(name string) error {
 	// set a task cookie and disable library fallback, or else libdevmapper will
 	// disable udev dm rules and delete the symlink under /dev/mapper by itself,
 	// no matter the removal is executed immediately or deferred by the kernel.
-	// Calling UdevWait on the cookie after task.run is not necessary, since deferred
-	// device removal is an asynchronous api call.
 	var cookie uint
 	var flags uint16
 	flags = DmUdevDisableLibraryFallback
-
 	if err := task.setCookie(&cookie, flags); err != nil {
 		return fmt.Errorf("devicemapper: Can not set cookie: %s", err)
 	}
+	defer UdevWait(&cookie)
 
```

Comment:
If you wait for cookie to complete, will we not wait till udev actually as removed the device and symlink? If yes, then deferred removal does not mean anything. If no, then it bags the question, what's the point of waiting on cookie. just to clean it up?